File: docs-examples/example-java/src/test/java/io/micronaut/nats/docs/jetstream/os/ObjectStoreHolder.java
package io.micronaut.nats.docs.jetstream.os;


import io.micronaut.context.annotation.Requires;

// tag::imports[]
import io.micronaut.nats.jetstream.annotation.ObjectStore;
import io.nats.client.JetStreamApiException;
import jakarta.inject.Inject;
import jakarta.inject.Singleton;

import java.io.IOException;
import java.io.InputStream;
import java.security.NoSuchAlgorithmException;
// end::imports[]

@Requires(property = "spec.name", value = "ObjectStoreTest")
// tag::clazz[]
@Singleton
public class ObjectStoreHolder {

    @Inject
    @ObjectStore("examplebucket")
    io.nats.client.ObjectStore store; // <1>

    public void put(String key, InputStream inputStream) throws JetStreamApiException, IOException, NoSuchAlgorithmException {
        store.put(key, inputStream);
    }

}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/nats/docs/jetstream/os/ObjectStoreHolder.kt
package io.micronaut.nats.docs.jetstream.os

import io.micronaut.context.annotation.Requires

// tag::imports[]
import io.micronaut.nats.jetstream.annotation.ObjectStore
import jakarta.inject.Inject
import jakarta.inject.Singleton
import java.io.InputStream

// end::imports[]

@Requires(property = "spec.name", value = "ObjectStoreSpec")
// tag::clazz[]
@Singleton
class ObjectStoreHolder {

    @Inject
    @field:ObjectStore("examplebucket")
    lateinit var store: io.nats.client.ObjectStore // <1>

    fun put(key: String, value:InputStream) {
        store.put(key, value)
    }

}
// end::clazz[]
